app/controllers/setting_controller.rb

```diff
@@ -244,4 +230,10 @@ def choose_weighted(weighted)
       target -= weight
     end
   end
+
+  # Samples a random number from the exponential distribution
+  def exponential(mean)
+    -mean * Math.log(rand()) if mean > 0
```

Comment:
What happens for mean >= 0?

Stupid question, but where does that formula come from?

Reply:
The function is from here: https://stackoverflow.com/a/18304464
I added a reference to the stackoverflow answer to the code.

The lambda parameter of the exponential distribution is only defined for >0 (see: https://en.wikipedia.org/wiki/Exponential_distribution#Probability_density_function).
Mean is the inverse of lambda and therefore should also be greater 0.
This function will return `nil` if mean is not greater 0.

Reply:
I just want to avoid a crashing consumer.

Couldn't you error-log situations where mean <= 0 and return a default of 1?

Reply:
I added an input validation for the consumer_per_minute setting.
In case consumer_per_minute is invalid (e.g. negative), it will be set to a default value and an error message is printed.
This will ensure that the calculated waiting time will always be a valid positive number.

Reply:
Then again, the `if mean > 0` check is no longer needed. ;)